### ISSUE TITLE
Forbid to have to many `except` cases

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#419]
# Rule request

## Thesis

We need to forbid too many `except` cases. This should be a configurable complexity rule.

## Reasoning

It does not make sense to have many `except` cases. It means that your code is not optimal, has many possible breaking point or is generally too complex.

**Comment by kxepal** [comment]
How many would be too many? Actually, each  except handles own case or there is sort of logic copy paste, which you unlikely could handle.

**Comment by kxepal** [comment]
For instance, I have such a function:
```
async def json_error_middleware(app, handler):
    async def middleware_handler(request):
        try:
            resp = await handler(request)
            if isinstance(resp, aiohttp.web.HTTPException):
                raise resp
        except aiohttp.web.HTTPNoContent:
            raise
        except aiohttp.web.HTTPBadRequest as exc:
            return error_response(400, "bad_request", exc.reason, exc.headers)
        except aiohttp.web.HTTPUnauthorized as exc:
            return error_response(401, "unauthorized", exc.reason, exc.headers)
        except aiohttp.web.HTTPForbidden as exc:
            return error_response(403, "forbidden", exc.reason, exc.headers)
        except aiohttp.web.HTTPNotFound as exc:
            return error_response(404, "not_found", exc.reason, exc.headers)
        except aiohttp.web.HTTPMethodNotAllowed as exc:
            return error_response(405, "not_allowed", exc.reason, exc.headers)
        except aiohttp.web.HTTPConflict as exc:
            return error_response(409, "conflict", exc.reason, exc.headers)
        else:
            return resp
```
Where I do dispatch upon exception type. Could it be refactored via sort of exception-to-str mapping? Yes, but that introduces some global variable thing or another long switch-case just for what? So is this fine? I think so.

**Comment by sobolevn** [comment]
I am not sure about the default value yet. `3` looks like a good place to start.

Considering your example I would refactor it like so:

```python
except aiohttp.web.HTTPException as exc:
    return error_response(exc.status_code, "conflict", exc.reason, exc.headers)
```

However, I am not sure that it is possible by default, but that's a problem of `aiohttp`. It is generally bad designed. 

So, making a mapping is a good thing in this case. Also, this case should be removed:

```python
except aiohttp.web.HTTPNoContent:
            raise
``` 

See https://github.com/wemake-services/wemake-python-styleguide/issues/366

**Comment by kxepal** [comment]
> Considering your example I would refactor it like so:

That wouldn't be right refactoring since I need to map each HTTPException subclass to some string.

> However, I am not sure that it is possible by default, but that's a problem of aiohttp. It is generally bad designed.

Well, that's not aiohttp issue at all. It's how my application returns responses for different HTTP errors. Replace JSON with some HTML pages - nothing much will change.

> So, making a mapping is a good thing in this case. 

But why I need standalone mapping, which is internal by fact, while `except`  does that mapping for free without creating useless one shot variables?

> Also, this case should be removed

Actually, not, since it goal to explicitly define behavior for exactly that kind of exception. Let's say, I would like to add `except Exception`case for 500 responses. So, HTTPNoContent will silently fall into this branch causing wrong behavior.
